support-frontend/test/selenium/util/PostDeployTestUser.scala

```diff
@@ -31,28 +34,56 @@ class PostDeployTestUserContribs(driverConfig: DriverConfig) extends TestUser {
 
 class PostDeployTestUserSubs(driverConfig: DriverConfig) extends TestUser {
 
-  private val client = new OkHttpClient()
-  private val requestToIdapi = new Request.Builder()
-    .url(Config.idapiNewTestUserUrl)
-    .addHeader(Config.idapiClientAccessTokenName, Config.idapiClientAccessTokenSecret)
-    .post(RequestBody.create(MediaType.parse("application/json"), "{}"))
-    .build()
-  private val response = client.newCall(requestToIdapi).execute()
-  private val responseBodyAsJson = parse(response.body().string()).getOrElse(Json.Null)
-  private val cursor = responseBodyAsJson.hcursor
-
   private def addTestUserCookies(testUsername: String) = {
     driverConfig.addCookie(name = "pre-signin-test-user", value = testUsername)
     driverConfig.addCookie(name = "_test_username", value = testUsername, domain = Some(Config.guardianDomain))
     driverConfig.addCookie(name = "_post_deploy_user", value = "true") // This enables the tests to use the mocked payment services
     driverConfig.addCookie(name = "GU_TK", value = "1.1") //To avoid consent banner, which messes with selenium
-    for (n <- 0 until 3) {
-      val cookie_name = cursor.downField("values").downN(n).downField("key").as[String].right.get
-      val cookie_value = cursor.downField("values").downN(n).downField("value").as[String].right.get
-      driverConfig.addCookie(name = cookie_name, value = cookie_value)
+    IdapiRequest.decodeCookies.foreach { cookie =>
+      driverConfig.addCookie(name = cookie.key, value = cookie.value)
     }
   }
 
   val username = "postDeployTestUser"
   addTestUserCookies(username)
 }
+
+case class IdapiCookie(key: String, value: String)
+
+object IdapiCookie {
+  implicit val decoder: Decoder[IdapiCookie] = deriveDecoder[IdapiCookie]
+}
+
+case class IdapiSuccessfulResponse(values: List[IdapiCookie])
+
+object IdapiSuccessfulResponse {
+  implicit val decoder: Decoder[IdapiSuccessfulResponse] = deriveDecoder[IdapiSuccessfulResponse]
+}
+
+object IdapiRequest {
+
+  private val client = new OkHttpClient()
+
+  private def logger = LoggerFactory.getLogger(this.getClass)
+
+  private val request: Request = new Request.Builder()
+    .url(Config.idapiNewTestUserUrl)
+    .addHeader(Config.idapiClientAccessTokenName, Config.idapiClientAccessTokenSecret)
+    .post(RequestBody.create(MediaType.parse("application/json"), "{}"))
+    .build()
+
+  private def getResult: Json = {
+    val response: Response = client.newCall(request).execute()
+    parse(response.body().string()).getOrElse(Json.Null)
+  }
+
+  def decodeCookies: List[IdapiCookie] = {
+    val responseDecoded: Either[DecodingFailure, IdapiSuccessfulResponse] = getResult.as[IdapiSuccessfulResponse]
+    var cookies = List[IdapiCookie]()
+    responseDecoded match {
+      case Left(l) => logger.error(s"Failed to decode IDAPI response: ${l}")
+      case Right(r) => cookies = r.values
+    }
+    cookies
```

Comment:
avoid using a var, and avoid hiding errors, if you can't handle them then just return a suitable failable type (either/try/option)
```suggestion
    responseDecoded match {
      case Left(error) => {
        logger.error(s"Failed to decode IDAPI response: $error")
        None
      }
      case Right(r) => Some(r.values)
    }
```

Reply:
That makes a lot of sense (I was trying to get it to do something similar but couldn't get the syntax right). Thank you!